dockerfile-image-update/src/main/java/com/salesforce/dockerfileimageupdate/utils/GithubAppCheck.java

```diff
@@ -119,41 +148,27 @@ protected boolean isGithubAppEnabledOnRepositoryWithGitApi(String fullRepoName)
      * Reference: https://github.com/mend/renovate-ce-ee/blob/main/docs/reporting-apis.md#repo-info
      */
+
     protected boolean isGithubAppEnabledOnRepositoryWithRenovateApi(String fullRepoName) {
+        return isGithubAppEnabledOnRepositoryWithRenovateApi(fullRepoName, httpClient);
+    }
+
+    protected boolean isGithubAppEnabledOnRepositoryWithRenovateApi(String fullRepoName, CloseableHttpClient httpClient) {
         try {
             String apiEndpoint = appServerApiEndpoint + "/api/repos/" + fullRepoName;
-            URL url = new URL(apiEndpoint);
-            HttpURLConnection conn = (HttpURLConnection) url.openConnection();
-            conn.setRequestMethod("GET");
-            conn.setRequestProperty("Accept", "application/json");
-            conn.setRequestProperty("authorization", "Bearer " + appServerApiToken);
-            conn.connect();
-            
-            Integer responseCode = conn.getResponseCode();
-            if (responseCode != 200) {
-                throw new RuntimeException("HttpResponseCode: " + responseCode);
-            } else {
-                String inline = "";
-                Scanner scanner = new Scanner(url.openStream());
-                while (scanner.hasNext()) {
-                    inline += scanner.nextLine();
-                }
-                scanner.close();
-
-                //Using the JSON simple library parse the string into a json object
-                JSONParser parse = new JSONParser();
-                JSONObject dataObject = (JSONObject) parse.parse(inline);
-                String appInstallationState = (String) dataObject.get("state");
-                String appActivationStatus = (String) dataObject.get("status");
-
-                // Return app installation and activation status
-                if (appInstallationState == "installed" && appActivationStatus == "activated") {
-                    return true;
+            HttpGet httpGet = new HttpGet(apiEndpoint);
+            httpGet.setHeader("Authorization", appServerApiToken);
+            httpGet.setHeader("Accept", "application/json");
+            try (CloseableHttpResponse response = httpClient.execute(httpGet)) {
+                int statusCode = response.getStatusLine().getStatusCode();
+                log.warn("[isGithubAppEnabledOnRepositoryWithRenovateApi] -- Response code `{}` while trying to get app installation by Renovate API", statusCode);
+                if (statusCode >= 500) {
+                    throw new IOException();
                 }
-                return false;
+                return statusCode == 200;
             }
         } catch (Exception e) {
-            e.printStackTrace();
+            log.warn("[isGithubAppEnabledOnRepositoryWithRenovateApi] -- Caught a IOException while trying to get app installation on repo {}. Exception: {}. Defaulting to False", fullRepoName, ExceptionUtils.getStackTrace(e));
             return false;
         }
```

Comment:
If you throw the IOException in the if statement it won't make it out of this method because you catching call exceptions here.  From `.retryExceptions(IOException.class)` the only retry condition is receiving an IOException outside of this method

Reply:
@cjneasbi Resolved